lib/src/cpp/common/LocatorList.hpp

```diff
@@ -345,7 +345,7 @@ void set_kind(
  */
 void set_port(
         utils::ParseXML& manager,
-        xercesc::DOMElement& xml_node,
+        xercesc::DOMNode* xml_node,
         const std::string& port,
         const std::string& index,
         const bool is_external);
```

Comment:
We should think carefully the needs of the API to define correctly the functions. In this case a pointer would be interesting so we can point to different nodes as we move deeper into the XML.